skyplane/api/client.py

```diff
@@ -142,3 +142,48 @@ def dataplane(
             return Dataplane(clientid=self.clientid, topology=topo, provisioner=self.provisioner, transfer_config=self.transfer_config)
         else:
             raise NotImplementedError(f"Dataplane type {solver_type} not implemented")
+
+    def download_object(self, bucket_name: str, provider: str, key: str, filename: str):
+        obj_store = ObjectStoreInterface.create(f"{provider}:infer", bucket_name)
+        obj_store.download_object(key, filename)
+
+    def upload_object(self, filename: str, bucket_name: str, provider: str, key: str):
+        obj_store = ObjectStoreInterface.create(f"{provider}:infer", bucket_name)
+        obj_store.upload_object(filename, key)
+
+    def exists(self, bucket_name: str, provider: str, key: str) -> bool:
+        obj_store = ObjectStoreInterface.create(f"{provider}:infer", bucket_name)
+        return obj_store.exists(key)
+
+    def bucket_exists(self, bucket_name: str, provider: str) -> bool:
+        # azure not implemented
+        if provider == "azure":
```

Comment:
Is there a reason we don't have Azure support yet? Would this design generalize?